src/ngraph_backend_manager.cc

```diff
@@ -0,0 +1,62 @@
+/*******************************************************************************
+ * Copyright 2018 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+
+#include "ngraph_backend_manager.h"
+
+#include <iostream>
+
+namespace tensorflow {
+namespace ngraph_bridge {
+
+BackendManager* BackendManager::Instance() {
+  static BackendManager instance;
+  return &instance;
+}
+
```

Comment:
It seems that `BackendManager` is supposed to be a singleton class.

In which case, can we make default/copy/assignment constructors/functions `private`

Reply:
It's usually a singleton class, but I've often found that it hurts testability to bake that knowledge into the class itself, beyond providing a Myers singleton accessor.  (I've added a comment to make it more clear what's going on)